crypto/evp_extra/evp_extra_test.cc

```diff
@@ -2674,3 +2697,71 @@ TEST_P(PerKEMTest, KAT) {
     EXPECT_EQ(Bytes(ss_expected), Bytes(ss));
   });
 }
+
+TEST_P(PerKEMTest, KeygenSeedTest) {
+  // ---- 1. Setup phase: generate a context and a key ----
+  // Create context of KEM type.
+  bssl::UniquePtr<EVP_PKEY_CTX> ctx(EVP_PKEY_CTX_new_id(EVP_PKEY_KEM, nullptr));
+  ASSERT_TRUE(ctx);
+
+  // Setup the context with specific KEM parameters.
+  ASSERT_TRUE(EVP_PKEY_CTX_kem_set_params(ctx.get(), GetParam().nid));
+
+  // Generate a key pair.
+  EVP_PKEY *raw = nullptr;
+  ASSERT_TRUE(EVP_PKEY_keygen_init(ctx.get()));
+
+  // ---- 2. Test getting the lengths only ----
+  size_t keygen_seed_len;
+  ASSERT_TRUE(EVP_PKEY_keygen_deterministic(ctx.get(), &raw, nullptr,
+                                            &keygen_seed_len));
+  EXPECT_EQ(keygen_seed_len, GetParam().keygen_seed_len);
+
+  // ---- 3. test failure mode on a seed len too small----
+  keygen_seed_len -= 1;
+  std::vector<uint8_t> small_keygen_seed(keygen_seed_len);
+  EXPECT_FALSE(EVP_PKEY_keygen_deterministic(ctx.get(), &raw,
+                                             small_keygen_seed.data(),
+                                             &keygen_seed_len));
+  EXPECT_EQ(EVP_R_INVALID_PARAMETERS, ERR_GET_REASON(ERR_peek_last_error()));
+
+  // ---- 3. test failure mode on a seed len too small----
+  keygen_seed_len += 2;
+  std::vector<uint8_t> big_keygen_seed(keygen_seed_len);
+  EXPECT_FALSE(EVP_PKEY_keygen_deterministic(ctx.get(), &raw, big_keygen_seed.data(),
+                                             &keygen_seed_len));
+  EXPECT_EQ(EVP_R_INVALID_PARAMETERS, ERR_GET_REASON(ERR_peek_last_error()));
+}
+
+TEST_P(PerKEMTest, EncapsSeedTest) {
+
+  // ---- 1. Setup phase: generate a context and a key ----
+  bssl::UniquePtr<EVP_PKEY_CTX> ctx;
+  ctx = setup_ctx_and_generate_key(GetParam().nid, nullptr, nullptr);
+  ASSERT_TRUE(ctx);
+
+  // ---- 2. Test getting the lengths only ----
+  size_t ct_len;
+  size_t ss_len;
+  size_t es_len;
+
+  ASSERT_TRUE(EVP_PKEY_encapsulate_deterministic(
+      ctx.get(), nullptr, &ct_len, nullptr, &ss_len, nullptr, &es_len));
+  EXPECT_EQ(ct_len, GetParam().ciphertext_len);
+  EXPECT_EQ(ss_len, GetParam().shared_secret_len);
+  EXPECT_EQ(es_len, GetParam().encaps_seed_len);
+
+  // ---- 3. test failure mode on a seed len too small----
+  es_len -= 1;
+  std::vector<uint8_t> ct(ct_len);
+  std::vector<uint8_t> ss(ss_len);
+  std::vector<uint8_t> es(es_len);
+
+  ASSERT_FALSE(EVP_PKEY_encapsulate_deterministic(
```

Comment:
this is actually an additional test, can we add:
```
// ---- 4. Test failure mode on seed being NULL ----
```

Reply:
fixed in [ad37a81](https://github.com/aws/aws-lc/pull/1578/commits/ad37a8166eadbd1af368220b672f7c951f142f44)